backend/app/datasets/service.py

```diff
@@ -74,6 +75,27 @@ def __init__(self, db: Session):
         self.db = db
         self.namespace_validator = NamespaceValidator(DatasetModel)
 
+    @staticmethod
+    def _build_prefix_tsquery(query: str) -> str | None:
```

Comment:
Can we add a couple of unit tests for this function, it's quite complex and that way we can ensure it does what is is supposed to

Reply:
Added 11 unit tests for this function